### DISCUSSION TITLE
Installation of PDF Exporter on a Polarion Windows Server - PDF Generation error

### DISCUSSION THREAD

**Discussion by sergePolarion** [SchweizerischeBundesbahnen/ch.sbb.polarion.extension.pdf-exporter#108] [Q&A] [answered]
Hi all,
I have installed the Polarion PDF exporter on my polarion windows server:

- Extension (.jar) installed under /polarion/polarion/extensions,  widget available then installed
- Python installed
- Weasyprint installed (i followed what is described here using pip (https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#windows), i read there is an executable but i don't know if we can use it 
- Stop/Start Polarion
Rest API enabled
-> i have the following error after 
Error occurred during PDF generation:
ch.sbb.polarion.extension.pdf.exporter.PdfExportException: Process (C:\Users\sergi\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\Scripts\weasyprint.exe -p -e utf8 -f pdf - -) exited with status code 1:

i would have liked to install weasyprint.exe in an other folder than /sergi/appData/Local/etc. but i didn't find how to do it (in case it is the cause)
You can see in the log file the 'french' error: le canal de communication a été fermé 
Any ideas
Thanks for your support
Serge
[log4j-errors-20240725-1203-02_PDF_Exporter.log](https://github.com/user-attachments/files/16375352/log4j-errors-20240725-1203-02_PDF_Exporter.log)

**Comment by grigoriev** [comment, marked as answer]
Hi @sergePolarion,

the preferable way to use WeasyPrint with PDF-Exporter is [WeasyPrint as Service in Docker](https://github.com/SchweizerischeBundesbahnen/ch.sbb.polarion.extension.pdf-exporter?tab=readme-ov-file#weasyprint-as-service-in-docker).

If you still want to use WeasyPrint as legacy CLI for Windows I would suggest installing WeasyPrint as [executable](https://github.com/Kozea/WeasyPrint/releases) binary:
<img width="669" alt="image" src="https://github.com/user-attachments/assets/f256ffdf-a3cb-4e0c-bb91-c4ec08035abd">

Best Regards,
Sergey

**Comment by sergePolarion** [comment]
Hi Sergey,
Thanks, for now on my demo server i will continue with CLI, i missed this part (i was reading the pdf, not the readme):, i think it is also required on windows:
To run WeasyPrint as a system's command-line tool specify following properties in file <POLARION_HOME>/**configuration**/polarion.properties:
ch.sbb.polarion.extension.pdf-exporter.weasyprint.connector=cli
ch.sbb.polarion.extension.pdf-exporter.weasyprint.executable=weasyprint
ch.sbb.polarion.extension.pdf-exporter.weasyprint.pdf.variant=pdf/a-2b
Use executable was my first thought, But how to tell the extension where to find the WeasyPrint executable ? 
Thanks
Serge

**Reply by grigoriev** [reply]
@sergePolarion,
 you can specify absolute path to WeasyPrint executable using `ch.sbb.polarion.extension.pdf-exporter.weasyprint.executable` property.

Another way is adding WeasyPrint executable to PATH.

**Comment by sergePolarion** [comment]
Thanks, 
It works now, i solved it by  adding /Python/scripts to the PATH,  copying weasyprint.exe (without venv) in this /scripts/ folder, and without polarion properties.
Best regards
Serge
